pkg/resourceinterpreter/default/native/retain.go

```diff
@@ -40,34 +41,62 @@ func retainPodFields(desired, observed *unstructured.Unstructured) (*unstructure
 		return nil, fmt.Errorf("failed to convert clusterPod from unstructured object: %v", err)
 	}
 
-	desiredPod.Spec.NodeName = clusterPod.Spec.NodeName
-	desiredPod.Spec.ServiceAccountName = clusterPod.Spec.ServiceAccountName
-	desiredPod.Spec.Volumes = clusterPod.Spec.Volumes
-	// retain volumeMounts in each container
-	for _, clusterContainer := range clusterPod.Spec.Containers {
-		for desiredIndex, desiredContainer := range desiredPod.Spec.Containers {
+	// Pod updates may not change fields other than `spec.containers[*].image`, `spec.initContainers[*].image`,
+	// `spec.activeDeadlineSeconds`, `spec.tolerations` (only additions to existing tolerations) or
+	// `spec.terminationGracePeriodSeconds` (allow it to be set to 1 if it was previously negative).
+	// More info: https://kubernetes.io/docs/concepts/workloads/pods/#pod-update-and-replacement
+	clusterPod.Labels = desiredPod.Labels
+	clusterPod.Annotations = desiredPod.Annotations
+	updatePodImages(desiredPod, clusterPod)
+	updatePodActiveDeadlineSeconds(desiredPod, clusterPod)
+	updatePodTolerations(desiredPod, clusterPod)
+
+	unstructuredObj, err := helper.ToUnstructured(clusterPod)
```

Comment:
The returned object is clusterPod, which is different from the processing logic in other places and may misunderstand other developers or code readers. Can we add some redundant processing to maintain the original logic habits?

Reply:
Or we can add additional comments to explain, my first reaction after seeing the code is whether the object is wrong, and I think @chaunceyjiang  also has a first impression misconception.

Reply:
Yeah, it's a good idea. There are many fields in _Pod_, but only a few fields that can be modified. I think it is unnecessary waste to copy from `clusterPod` to `desiredPod`.